src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -2167,6 +2167,197 @@ protected override void BeginProcessing()
+        [Alias("ServiceName", "SN")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The following is the definition of the input parameter "InputObject".
+        /// Specifies ServiceController object representing the services to be removed.
+        /// Enter a variable that contains the objects or type a command or expression
+        /// that gets the objects.
+        /// </summary>
+        [Parameter(ValueFromPipeline = true, ParameterSetName = "InputObject")]
+        public ServiceController InputObject { get; set; }
+
+        /// <summary>
+        /// The following is the definition of the input parameter "ComputerName".
+        /// Set the properties of service running on the list of computer names
+        /// specified. The default is the local computer.
+        /// Type the NETBIOS name, an IP address, or a fully-qualified domain name of
+        /// one or more remote computers. To indicate the local computer, use the
+        /// computer name, "localhost" or a dot (.). When the computer is in a different
+        /// domain than the user, the fully-qualified domain name is required.
+        /// </summary>
+        [Parameter(ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        [Alias("cn")]
+        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
+        public String[] ComputerName { get; set; } = new string[] { "." };
+
+        #endregion Parameters
+
+        #region Overrides
+        /// <summary>
+        /// Remove the service
+        /// </summary>
+        [ArchitectureSensitive]
+        protected override void ProcessRecord()
+        {
+            ServiceController service = null;
+            string serviceComputerName = null;
+            foreach (string computer in ComputerName)
+            {
+                bool objServiceShouldBeDisposed = false;
+                try
+                {
+                    if (_ParameterSetName.Equals("InputObject", StringComparison.OrdinalIgnoreCase) && InputObject != null)
+                    {
+                        service = InputObject;
+                        Name = service.ServiceName;
+                        serviceComputerName = service.MachineName;
+                        objServiceShouldBeDisposed = false;
+                    }
+                    else
+                    {
+                        serviceComputerName = computer;
+                        service = new ServiceController(Name, serviceComputerName);
+                        objServiceShouldBeDisposed = true;
+                    }
+                    Diagnostics.Assert(!String.IsNullOrEmpty(Name), "null ServiceName");
+                    // "new ServiceController" will succeed even if
+                    // there is no such service.  This checks whether
+                    // the service actually exists.
```

Comment:
Should we put the comment before Diagnostics.Assert?
And please reformat the comment  - we can use wider lines.